### DISCUSSION TITLE
Question about update from v11.x to v12.x and token retrieval with multiple instance

### DISCUSSION THREAD

**Discussion by manuquentin** [invertase/react-native-firebase#6145] [Q&A] [answered]
Hi !
We're currently using `@react-native-firebase/messaging` `v11.5.0` to retrieve the device token for a firebase application.

We have different customers and we set `authorizedEntity` in the `getToken()` method with their project id to fetch a token corresponding to their firebase application.

But [with new refactoring about iid](https://github.com/invertase/react-native-firebase/commit/aba8fc643f529d8cab32aeefaf5598b1e8a82335#diff-945d05be567b9871eff943790a8b6ff7efde8bcd48a0b41d24d240b4687859c0L119-R118) this parameter has been removed and we should now use `Installations` as I understand.

Looking at the [`installations package`](https://github.com/invertase/react-native-firebase/tree/main/packages/installations) I don't see how I can change this `authorizedEntity`, the [migration guide](https://firebase.google.com/docs/projects/manage-installations#fid-iid) doesn't mention it neither.

How can I retrieve a device token for another `authorizedEntity` with `react-native-firebase` recent version ?

Thanks !

**Comment by mikehardy** [comment, marked as answer]
@manuquentin I apologize - but I'm not sure it is possible. Scopes for tokens are simply gone as far as I know. This may be unsatisfying as an answer here but I think the best course of action is either to ping the firebase folks upstream at firebase-android-sdk or on stackoverflow with correct firebase tags (they offer support there). They may have a mechanism to do it or a suggested workaround, otherwise the only thing I can think of is to change perspective to really just think of the installation ID as "the app on the device" and if you need to differentiate further to maintain that information in your own backend so you may target devices correctly

**Reply by mikehardy** [reply]
Hey! It appears there is a possibility to do this, And @ecexplorer has discovered how to add it here #6173 in Android at least - follow that issue
